flink-yarn/src/main/java/org/apache/flink/yarn/AbstractYarnClusterDescriptor.java

```diff
@@ -409,6 +411,12 @@ public void terminateCluster(ApplicationId applicationId) throws FlinkException
 		}
 	}
 
+	private void checkConfig(ClusterSpecification clusterSpecification) {
```

Comment:
This should throw a checked exception. Something like `InvalidConfigurationException` or so. But not just fail with an `IllegalStateException`.